.github/workflows/test.yml

```diff
@@ -0,0 +1,24 @@
+# Runs pytest on the matrix of supported platforms any Python versions.
+name: Test
```

Comment:
[consider][minor] Perhaps worth a concurrency group, since max concurrent jobs [seems to be a per-account setting](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits)

> The number of concurrent jobs you can run in **your account** depends on 
your GitHub plan, as indicated in the following table. If exceeded, any 
additional jobs are queued.

GitHub plan | Total concurrent jobs | Maximum concurrent macOS jobs
-- | -- | --
Free | 20 | 5

Minor though, it's unlikely we'll have lots of concurrent PRs to this repo and others.

Reply:
Good idea, added [here](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/3/commits/37368c1c7d1348d46c13096e54b59f8aabe22730).